iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/pointer_repository.inl

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_UTILS_RELOCATABLE_POINTER_POINTER_REPOSITORY_INL
+#define IOX_UTILS_RELOCATABLE_POINTER_POINTER_REPOSITORY_INL
+
+#include "iceoryx_utils/internal/relocatable_pointer/pointer_repository.hpp"
+
+namespace iox
+{
+namespace rp
+{
+template <typename id_t, typename ptr_t, uint64_t CAPACITY>
+inline PointerRepository<id_t, ptr_t, CAPACITY>::PointerRepository() noexcept
+    : m_info(CAPACITY)
+{
+}
+
+template <typename id_t, typename ptr_t, uint64_t CAPACITY>
+inline bool PointerRepository<id_t, ptr_t, CAPACITY>::registerPtr(id_t id, ptr_t ptr, uint64_t size) noexcept
+{
+    if (id > MAX_ID)
+    {
+        return false;
+    }
+    if (m_info[id].basePtr == nullptr)
+    {
+        m_info[id].basePtr = ptr;
+        m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uintptr_t>(ptr) + size - 1U);
+        if (id > m_maxRegistered)
+        {
+            m_maxRegistered = id;
+        }
+        return true;
+    }
+    return false;
+}
+
+template <typename id_t, typename ptr_t, uint64_t CAPACITY>
+inline id_t PointerRepository<id_t, ptr_t, CAPACITY>::registerPtr(const ptr_t ptr, uint64_t size) noexcept
```

Comment:
We should use RAII here as well and return a `cxx::unique_ptr<id_t>` which calls unregisterPtr when it goes out of scope.

Reply:
added to issue

Reply:
PointerRepository practically internal API (implementation detail) and optimized for speed (could still be improved in this regard but this would slow it down).
This behavior is also not what we want, otherwise I need to keep those ids around to avoid unregistering. Not cool.

Reply:
@MatthiasKillat 
1. If you do not want to keep this ids around how do you keep track when cleaning up everything with `unregisterPtr`?
2. Is this on our hot path registering (and unregistering) memory? I thought this is only called on startup or whenever you map a new shared memory into your process?!